Failed to push a comment in case of push event

https://github.com/aquaproj/example-update-checksum/actions/runs/3963836709/jobs/6792117797

```
Error: aqua-checksums.json isn't latest. A commit is pushed automatically to update aqua-checksums.json.
time="2023-01-20T01:15:00Z" level=info msg="download and unarchive the package" aqua_version=1.32.0 env=linux/amd64 exe_name=ghcp exe_path=/home/runner/.local/share/aquaproj-aqua/pkgs/github_release/github.com/int128/ghcp/v1.13.2/ghcp_linux_amd64.zip/ghcp package=int128/ghcp package_name=int128/ghcp package_version=v1.13.2 program=aqua registry=standard
01:15:01.334831 INFO  Author and committer: aqua-update-checksum-renovatepush[bot]
01:15:01.334887 INFO  Updating the branch (main) by fast-forward
01:15:01.44[97](https://github.com/aquaproj/example-update-checksum/actions/runs/3963836709/jobs/6792117797#step:5:102)63 INFO  Uploaded aqua-checksums.json as blob c6f21a2f3[98](https://github.com/aquaproj/example-update-checksum/actions/runs/3963836709/jobs/6792117797#step:5:103)30cff6af6bb331e1d1fd6e66185f0
01:15:01.604908 INFO  Created tree 38bcf15c3ef092dcc5454a4e735de2800d2fa9d9
01:15:01.857350 INFO  Created commit 42bcb072ef34ac922516e5ea26849f3b188a2ab4
01:15:02.062538 INFO  Created a commit with 1 changed file(s)
Error: could not commit the files: could not update the existing branch (main): error while updating main branch: GitHub API error: Branch is protected: Waiting on code owner review from mini-core and/or suzuki-shunsuke. Required status check "status-check" is expected.
Error: Process completed with exit code 1.
```

This action tried to push the main branch. The branch was wrong.

https://github.com/aquaproj/update-checksum-action/blob/c6ac4f2845d2f2f4865a3566110a8f1d216fa0a6/action.yaml#L43-L45

`$GITHUB_HEAD_REF`

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

> GITHUB_HEAD_REF
> The head ref or source branch of the pull request in a workflow run. This property is only set when the event that triggers a workflow run is either pull_request or pull_request_target. For example, feature-branch-1.

--- a/action.yaml
+++ b/action.yaml
@@ -40,7 +40,11 @@ runs:
         if ! git diff --cached --exit-code ${CHECKSUM_FILE}; then
           echo "::error file=${CHECKSUM_FILE}::${CHECKSUM_FILE} isn't latest. A commit is pushed automatically to update ${CHECKSUM_FILE}."
           # https://github.com/int128/ghcp
-          ghcp commit -r "$GITHUB_REPOSITORY" -b "$GITHUB_HEAD_REF" \
+          branch=${GITHUB_HEAD_REF:-}
+          if [ -z "$branch" ]; then
+            branch=$GITHUB_REF
+          fi
+          ghcp commit -r "$GITHUB_REPOSITORY" -b "$branch" \
             -m "chore(aqua): update $CHECKSUM_FILE" \
             aqua-checksums.json
           exit 1

--- a/action.yaml
+++ b/action.yaml
@@ -42,7 +42,7 @@ runs:
           # https://github.com/int128/ghcp
           branch=${GITHUB_HEAD_REF:-}
           if [ -z "$branch" ]; then
-            branch=$GITHUB_REF
+            branch=$GITHUB_REF_NAME
           fi
           ghcp commit -r "$GITHUB_REPOSITORY" -b "$branch" \
             -m "chore(aqua): update $CHECKSUM_FILE" \

